typicode/lowdb
# chore: add config to support cjs

### Issue

closes https://github.com/typicode/lowdb/issues/471

***

### Description 

This PR adds backwards compatibility to `lowdb` for `commonjs`.

### To Test

1. Reproduce error in a node project e.g. `express.js`
2. In this project, run the following command to install the version of this PR branch with access to project's `lib` directory.

   ```sh
   npm install github:dankreiger/lowdb#chore/esm-cjs-hybrid-WITH-LIB
   ```

2. Error should be fixed in https://github.com/typicode/lowdb/issues/471


***

### New Build Structure

```sh
lib/
├── cjs
│   ├── Low.js
│   ├── LowSync.js
│   ├── MissingAdapterError.js
│   ├── adapters
│   │   ├── JSONFile.js
│   │   ├── JSONFileSync.js
│   │   ├── LocalStorage.js
│   │   ├── Memory.js
│   │   ├── MemorySync.js
│   │   ├── TextFile.js
│   │   └── TextFileSync.js
│   ├── index.js
│   └── package.json
├── esm
│   ├── Low.js
│   ├── LowSync.js
│   ├── MissingAdapterError.js
│   ├── adapters
│   │   ├── JSONFile.js
│   │   ├── JSONFileSync.js
│   │   ├── LocalStorage.js
│   │   ├── Memory.js
│   │   ├── MemorySync.js
│   │   ├── TextFile.js
│   │   └── TextFileSync.js
│   ├── index.js
│   └── package.json
└── types
    ├── Low.d.ts
    ├── LowSync.d.ts
    ├── MissingAdapterError.d.ts
    ├── adapters
    │   ├── JSONFile.d.ts
    │   ├── JSONFileSync.d.ts
    │   ├── LocalStorage.d.ts
    │   ├── Memory.d.ts
    │   ├── MemorySync.d.ts
    │   ├── TextFile.d.ts
    │   └── TextFileSync.d.ts
    └── index.d.ts
 ```

### Note

- Sometimes I choose to do this with `rollup.js` or `webpack`, but I noticed this project is trying to stay slim, so I just stuck with the `typescript` build process.
- The separate builds inherit their respective module format by referencing the specified `type` in the nearest parent/sibling `package.json`. 
    - Source: https://nodejs.org/api/packages.html#packages_type.
